### ISSUE TITLE
Wider range of track and amenities detail levels / Smarter revealing and hiding

### ISSUE THREAD

**Issue by m-h-t** [andreynovikov/trekarta#65]
I find that often, when zooming out a bit, hiking tracks tracks and amenities disappear too quickly from the map. This means that I can only see detailed section of a hike, but not the complete track I'm planning on hiking. Have a look at these screenshots:

<img src="https://user-images.githubusercontent.com/3340888/52514887-d4822d00-2c69-11e9-8cc9-8a0fc45cff4e.png" height="600">`

Only slightly more zoomed out, all amenities and more importantly the track itself disappears:

<img src="https://user-images.githubusercontent.com/3340888/52514890-dc41d180-2c69-11e9-9f67-e56fa6e5c727.png" height="600">`

I think it would be good to be able to view hiking tracks at a zoom level corresponding to a couple of days worth of hiking. Like, as a minimum:

<img src="https://user-images.githubusercontent.com/3340888/52515039-27101900-2c6b-11e9-804a-80acedcba7a1.png" height="600">`

Also, all amenities of a specific type disappear at once. Perhaps it would be possible to hide them individually according to relevance (not sure what the data sources offer with regards to this), or maybe better, cluster them? Also, if an area is not cluttered, I think amenities don't need to be faded out at all, as long as they don't collide with any other marking.

**Comment by andreynovikov** [comment]
You can configure the zoom level for different types of amenities through maps advanced menu (long tap on maps button).

As for tracks it would be not good when there is a high density of tracks (see Alps or Greece for example). This will produce a clutter on the map in such location. This can be made configurable but I have to think about it.

**Comment by m-h-t** [comment]
Hi @andreynovikov 

I forgot to mention - the zoom level in the map advanced menu is already set to maximum for these amenities. Maybe the range of the slider could be extended, so that the maximum level is higher than what it currently is?

Yeah, I realise that this might be problematic for areas that have a more dense network of tracks / amenities, but I think only selectively hiding them when the map actually becomes too cluttered for the area currently viewed might be very beneficial.

**Comment by andreynovikov** [comment]
No, currently it works just like filters - all information about amenities is encoded in the map tiles on zoom levels from 14 to 17. Zoom level 13 does not contain any amenities so they can not be technically made visible.

In #60 I plan to remove amenities from map tile data and store them separately. This will make it possible to show them on any map zoom.

**Comment by m-h-t** [comment]
Alright, I understand what you mean.

That would be really good and very helpful for a number of regions. Thank you for your work :-)
